pkg/proxy/proxieriptables.go

```diff
@@ -0,0 +1,696 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package proxy
+
+/*
+NOTE: this needs to be tested in e2e since it uses iptables for everything.
+*/
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/base64"
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/types"
+	"k8s.io/kubernetes/pkg/util"
+	utilexec "k8s.io/kubernetes/pkg/util/exec"
+	"k8s.io/kubernetes/pkg/util/iptables"
+)
+
+// NOTE: It will be tricky to determine the exact version needed.
+// Some features will of course be backported in various distros and this could get pretty hairy.
+// However iptables-1.4.0 was released 2007-Dec-22 and appears to have every feature we use,
+// so this seems prefectly reasonable for now.
+const (
+	IPTABLES_MIN_MAJOR = 1
+	IPTABLES_MIN_MINOR = 4
+	IPTABLES_MIN_PATCH = 0
+)
+
+func ShouldUseProxierIptables() (bool, error) {
+	exec := utilexec.New()
+	major, minor, patch, err := iptables.GetIptablesVersion(exec)
+	if err != nil {
+		return false, err
+	}
+	return (major >= IPTABLES_MIN_MAJOR) && (minor >= IPTABLES_MIN_MINOR) && (patch >= IPTABLES_MIN_PATCH), err
```

Comment:
Though 3 digit number probably isn't safe either, as i've seen multidigit patch versions.
We may want a different metric here anyhow, but in the meantime i will do:
 `if (major > IPTABLES_MIN_MAJOR) || (major == IPTABLES_MIN_MAJOR && ...`